src/main/java/com/google/cloud/spanner/pgadapter/ConnectionHandler.java

```diff
@@ -743,14 +756,21 @@ && getServer().getOptions().shouldAutoDetectClient()) {
    */
   public void maybeDetermineWellKnownClient(ParseMessage parseMessage) {
     if (!this.hasDeterminedClientUsingQuery) {
-      if (this.wellKnownClient == WellKnownClient.UNSPECIFIED
-          && getServer().getOptions().shouldAutoDetectClient()) {
-        setWellKnownClient(ClientAutoDetector.detectClient(parseMessage));
+      if (parseMessage.getStatement().getStatementType() == StatementType.CLIENT_SIDE
+          && skippedAutoDetectParseMessages.size() < 10) {
```

Comment:
How is number 10 decided?

Reply:
A relatively random choice. This is the maximum number of messages that we will buffer before forcing a client detection. This is just a safety measure against clients that might send a lot of random client-side statements at the beginning. I'll add a comment to the code to explain that.